includes/islandora_multi_batch.inc

```diff
@@ -806,29 +806,44 @@ class IslandoraMultiBatchObject extends IslandoraBatchObject {
                   error_log('can not open ' . $this->preprocessorParameters['zipfile'] . ' with code:'. (int)$opened);
                 }
               }
+
+              // This will handle all cases, except ZIP.
+
               else {
-                // This will handle all cases, except ZIP.
                 $fileurl = islandora_multi_importer_remote_file_get(trim($this->objectInfo['data'][$method[1]]));
                 if ($fileurl) {
                   $themime = $this->getMimetype(drupal_basename($fileurl));
-                      if ($dsid == "HOCR" && $themime == "application/octet-stream") {
-                          $themime = "text/html";
-                      }
-                      if ($dsid == "OCR" && $themime == "application/octet-stream") {
-                          $themime = "text/plain";
-                      }
+                  if ($dsid == "HOCR" && $themime == "application/octet-stream") {
+                    $themime = "text/html";
+                  }
+                  if ($dsid == "OCR" && $themime == "application/octet-stream") {
+                    $themime = "text/plain";
+                  }
+
+                  // $file = system_retrieve_file($fileurl, $tmpName, FALSE, FILE_EXISTS_RENAME);
+
+                  $file = (object) array(
```

Comment:
I feel maybe your hook should deal with the correct format return, this could be probably an overkill if your hook is not invoked. ```islandora_multi_importer_remote_file_get()``` (without a hook) will invoke
system_retrieve_file($url, $local, FALSE, FILE_EXISTS_REPLACE); will return the following:
```
If it succeeds and $managed is FALSE, the location where the file was saved.
If it fails, FALSE.
```

But never 
```
If it succeeds and $managed is TRUE, a \Drupal\file\FileInterface object which describes the file.
```

And https://github.com/mnylc/islandora_multi_importer/pull/70/files#diff-d34e58beef14ce9ff000b83cbec66205R573 deals with the already set 'datastream_file' => $fileurl, perfectly fine.

Maybe i'm missing a need here and open to discuss the use case of course

Reply:
Not sure I understand this entirely, but in my attempts this simply didn't work.  I would really welcome a phone call to discuss because this portion has me confused too.